### ISSUE TITLE
questions about some details of code

### ISSUE THREAD

**Issue by zhyx12** [huitangtang/SRDC-CVPR2020#1]
Thanks for your work, I'm reading the code and get confused about some details:

1. For the usage of k_means, kernel_k_means, and spherical_k_means in main() function, change_target is set to False, so that the target label is not assigned, will it influence the training of network since TarDisClusterLoss() requires target label from train_loader_target?

2. I don't find the corresponding codes for Eq (2) in the paper which calculates the auxiliary distribution Q. It seems that Q in Eq (2) comes from k_means method which is based on features not final probability p.

**Comment by huitangtang** [comment]
Glad to hear from you. 

For the first question, we only use the assigned pseudo labels by k-means clustering in the feature space before the first FC layer. Thus, we set change_target to False for k-means clustering in the feature space before the second FC layer.

For the second question, we implement Eq. (2) in lines 145-146 in TarDisClusterLoss(). 

Please check it carefully.

Wish you all the best.
